src/backend/utils/adt/ri_triggers.c

```diff
@@ -430,11 +435,11 @@ RI_FKey_check(TriggerData *trigdata)
 	 * referenced side.  The reason is that our snapshot must be fresh in
 	 * order for the hack in find_inheritance_children() to work.
 	 */
-	ri_PerformCheck(riinfo, &qkey, qplan,
+	RUN_AS_PSQL(ri_PerformCheck(riinfo, &qkey, qplan,
```

Comment:
instead of changing all the calls, can we change inside the function at lowest level where it is required?